ggml/src/ggml-cpu/ggml-cpu-quants.c

```diff
@@ -5175,83 +5109,174 @@ void ggml_vec_dot_q2_K_q8_K(int n, float * GGML_RESTRICT s, size_t bs, const voi
 #elif defined __riscv_v_intrinsic
 
     float sumf = 0;
-    uint8_t temp_01[32] = {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
-                            1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1};
-
-    for (int i = 0; i < nb; ++i) {
 
-        const uint8_t * q2 = x[i].qs;
-        const  int8_t * q8 = y[i].qs;
-        const uint8_t * sc = x[i].scales;
+    if (__riscv_vlenb() >= 32) {
```

Comment:
I assume this is a system call? We should not have these at such deep level of computation. See how we have resolved this in the Arm kernels.

Reply:
> I assume this is a system call?

`__riscv_vlenb()` is not a system call. It will be compiled into a single `vsetvli rd, zero, e8, m1` instruction, similar to `svcntb()` in the Arm kernels.

> See how we have resolved this in the Arm kernels.

I'll restructure my code in revision.

Reply:
> I'll restructure my code in revision.

In that case, it can stay as it is. I thought it might have some performance implications.